src/npc.cpp

```diff
@@ -1783,6 +1787,30 @@ Creature::Attitude npc::attitude_to( const Creature &other ) const
     return A_NEUTRAL;
 }
 
+void npc::npc_dismount()
+{
+    tripoint pnt;
+    for( const auto &elem : g->m.points_in_radius( pos(), 1 ) ) {
+        if( g->is_empty( elem ) ) {
+            pnt = elem;
+            break;
+        }
+    }
+    if( pnt != tripoint_zero && mounted_creature ) {
```

Comment:
Checking for a `mounted_creature` should be at the top of the function as *all* the remaing code (including the search for a suitable place) can be skipped than.